services/kvstoremanager/redis/redis.go

```diff
@@ -169,10 +174,46 @@ func (m *RedisManager) HSet(hash, key string, value interface{}) (err error) {
 	return err
 }
 
-func (m *RedisManager) ExtractJSONSetArgs(jsonData json.RawMessage) ([]string, error) {
-	key := gjson.GetBytes(jsonData, "message.key").String()
-	path := gjson.GetBytes(jsonData, "message.path").String()
-	jsonVal := gjson.GetBytes(jsonData, "message.value")
+type NonRootInsertReturn struct {
+	SetArgsPath string
+	MergeTo     string
+	MergeFrom   interface{}
+}
+
+func (m *RedisManager) setArgsForMergeStrategy(inputArgs setArguments) ([]string, error) {
```

Comment:
should we have return arguments at struct rather than string array for better maintainability

Reply:
Addressed